sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -663,7 +663,16 @@ long flush(
                   retrieveErrorDetails(contexts));
               failedContext.failureCount += 1;
 
-              invalidateWriteStream();
+              Throwable error = Preconditions.checkStateNotNull(failedContext.getError());
```

Comment:
Line 607 and retrieveErrorDetails seems to expect this is null sometimes, perhaps safer to just set quotaError to false if getError is null?

Reply:
done